runtimes/common/src/dot_names/origin.rs

```diff
@@ -0,0 +1,83 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+pub(super) mod dot_names {
+	use frame_support::{pallet_prelude::OptionQuery, storage_alias, traits::PalletInfoAccess};
+	use sp_core::Get;
+	use sp_std::marker::PhantomData;
+
+	use crate::AccountId;
+
+	const LOG_TARGET: &str = "runtime::DotNames::AllowedDotNameClaimer";
+
+	// TODO: When upgrading to 1.8.0, which introduces a new `pallet-parameteres`,
```

Comment:
I see a big warning, that the pallet is not audited, and should not be used in production. 

Could you also maybe elaborate, how the pallet would work to fix this problem? As far as I can see, the pallet is used to update a parameter of another pallet. Why do we need that?

Reply:
The pallet is basically a generic key-value store where each key has an authorised origin that can change it. So in our case the value we need could be stored in there and updated by governance. I did not read about production readiness tho.

Reply:
I am struggling to get why we need the new pallet. You have already described the process in the description of the PR

Reply:
The main limitation of this solution is that we don't have any try-runtime support, and it's effectively a storage element the runtime has no knowledge about, meaning it won't be checked in any sorts of tests. Using a pallet would give us a "standard" place to store all our stuff, and to migrate and check accordingly.

Reply:
I see